server/server.go

```diff
@@ -0,0 +1,110 @@
+// Package server provides a server implementation to connect network transport
+// protocols and service business logic by defining server endpoints.
+package server
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	microserver "github.com/giantswarm/microkit/server"
+	"github.com/giantswarm/micrologger"
+	"github.com/spf13/viper"
+
+	"github.com/giantswarm/extservice-operator/server/endpoint"
+	"github.com/giantswarm/extservice-operator/service"
+)
+
+type Config struct {
+	Logger  micrologger.Logger
+	Service *service.Service
+	Viper   *viper.Viper
+
+	ProjectName string
+}
+
+func New(config Config) (microserver.Server, error) {
+	var err error
+
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
```

Comment:
Nitpick but I prefer the `microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)` style so you also get the type of the config struct.

If we are going to use that then we should use it everywhere IMO.